lib/sles4sap.pm

```diff
@@ -245,6 +245,7 @@ sub copy_media {
     # Then verify everything was copied correctly
     # NOTE: checksum is generated with this command: "find . -type f -exec md5sum {} \; > checksums.md5sum"
     my $chksum_file = 'checksum.md5sum';
+    return 1 if (! -f $chksum_file);
```

Comment:
Thanks for adding this test :-)

Just 2 comments:
- `-e` is more generic and better here than `-f`, but just my 2 cents
- could be also good to `die` here with an explicit message instead of a return, we could think that the copy itself failed, and that's not exactly true in that case

Reply:
This line of code just skip the rest of the function if the checksum file doesn't exist.

Reply:
But we definitely don't want this! At least could be done for testing purposes but not in production! (and for testing purposes we could skip the `md5sum` check if you want to speed up the process).

I know that it's a WIP PR, but if this code is a temporal workaround please add a comment.

Reply:
IMO, if the checksum file doesn't exist, we should create it on the expected spot so it can be used for the next tests.

Reply:
I'm copying a subdirectory of HANA where this file doesn't exist.  I think we should only give a warning if the checksum file doesn't exist.

Reply:
Oh, yes right. That makes sense.

Reply:
In that case why keeping this? The goal of this is to ensure that the integrity of the copied media is correct.

> I'm copying a subdirectory of the HANA media where this file doesn't exist.

Why? HANA should be entirely copied. If it's for the speedup process discussed recently please add a variable such as `SPEEDUP_COPY` or something similar set to `0` by default and use it to copy only some part or not check the media in that case.

Reply:
Thinking more about this, I think:

- A variable to control whether `md5sum -c` is run or not. By default, the check is done.
- If said variable is not set, `md5sum -c` is always run, either by using the checksum file present in the share, or by generating a local checksum file from the contents of the share.
- Generated checksum files are not saved in the share.

Just my 0.02.

Reply:
> * If said variable is not set, `md5sum -c` is always run, either by using the checksum file present in the share, or by generating a local checksum file from the contents of the share.

I will add a variable but generating a local checksum in the test code doesn't make sense as these files are copied from NFS/SMB anyway.